services/federator/src/Federator/Validation.hs

```diff
@@ -43,12 +52,62 @@ federateWith targetDomain = do
     AllowAll -> True
     AllowList (AllowedDomains domains) -> targetDomain `elem` domains
 
--- | Validates an unknown domain string against the allowList using the federator startup configuration
-validateDomain :: Members '[Polysemy.Reader RunSettings, Polysemy.Error InwardError] r => Text -> Sem r Domain
-validateDomain unparsedDomain = do
+decodeCertificate ::
+  Members '[Polysemy.Error InwardError] r =>
```

Comment:
Just nitpicking: given that there is only one effect in the list, you can use `Member` instead of `Members`.

Reply:
Fixed in acec8903fea4b886e98d415a6418a9e908ef4f55.